packages/devextreme/js/__internal/scheduler/m_utils_time_zone.ts

```diff
@@ -55,17 +61,106 @@ const getDaylightOffset = (startDate, endDate) => new Date(startDate).getTimezon
 
 const getDaylightOffsetInMs = (startDate, endDate) => getDaylightOffset(startDate, endDate) * toMs('minute');
 
-const calculateTimezoneByValue = (timezone, date = new Date()) => {
-  // NOTE: This check could be removed. We don't support numerical timezones
-  if (typeof timezone === 'string') {
-    const dateUtc = createUTCDate(date);
-    return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+const isValidDate = (date: Date) => date instanceof Date && !isNaN(date.valueOf());
+
+const calculateTimezoneByValueOld = (timezone: string, date = new Date()) => {
+  const customTimezones = timeZoneDataUtils.getTimeZonesOld();
+  if (customTimezones.length === 0) {
+    return undefined;
+  }
+
+  const dateUtc = createUTCDate(date);
+  return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+};
+
+const calculateTimezoneByValueCore = (timeZone: string, date = new Date()): number | undefined => {
+  let offset;
+  try {
+    offset = getStringOffset(timeZone, date);
+  } catch (e) {
+    errors.log('W0009', timeZone);
+    return undefined;
+  }
+
+  if (offset === GMT) {
+    return 0;
   }
-  return timezone;
+
+  const isMinus = offset.substring(3, 4) === '-';
+  const hours = offset.substring(4, 6);
+  const minutes = offset.substring(7, 9);
+
+  const result = parseInt(hours, 10) + parseInt(minutes, 10) / MINUTES_IN_HOUR;
+  return isMinus ? -result : result;
+};
+
+const calculateTimezoneByValue = (timeZone: string | undefined, date = new Date()): number | undefined => {
+  if (!timeZone) {
+    return undefined;
+  }
+  const isValidTimezone = timeZoneList.value.includes(timeZone);
+  if (!isValidTimezone) {
+    errors.log('W0009', timeZone);
+    return undefined;
+  }
+
+  if (!isValidDate(date)) {
+    return undefined;
+  }
+
+  let result = calculateTimezoneByValueOld(timeZone, date);
+  if (result === undefined) {
+    result = calculateTimezoneByValueCore(timeZone, date);
+  }
+  return result;
+};
+
+// 'GMT±XX:YY' or 'GMT' format
+const getStringOffset = (timeZone: string, date = new Date()): string => {
+  const dateTimeFormat = new Intl.DateTimeFormat('en-US', {
+    timeZone,
+    timeZoneName: 'longOffset',
+  } as any);
+
+  const result = dateTimeFormat.formatToParts(date)
+    .filter((part) => part.type === 'timeZoneName')
+    .map((p) => p.value)[0];
+
+  const isSupportedFormat = offsetFormatRegexp.test(result);
+  if (!isSupportedFormat) {
+    errors.log('W0009', timeZone);
+  }
+
+  return result;
+};
```

Comment:
Let's change this function to handle all possible unhandled errors:
```suggestion
const getStringOffset = (timeZone: string, date = new Date()): string => {
  if (typeof Intl?.DateTimeFormat !== 'function') {
    return undefined;
  }

  const dateTimeFormat = new Intl.DateTimeFormat('en-US', {
    timeZone,
    timeZoneName: 'longOffset',
  } as any);
  const result = dateTimeFormat?.formatToParts?.(date)
    .find(({ type }) => type === 'timeZoneName')?.value ?? '';

  const isSupportedFormat = offsetFormatRegexp.test(result);

  return isSupportedFormat ? result : undefined;
};
```